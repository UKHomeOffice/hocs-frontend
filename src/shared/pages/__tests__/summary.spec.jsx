import React from 'react';
import Summary from '../summary.jsx';
import { MemoryRouter } from 'react-router-dom';

const mockCaseData = {
    uuid: 'CASE_UUID',
    type: 'CASE_TYPE',
    reference: 'CASE_REF',
    timestamp: 'CASE_TIMESTAMP',
    stages: [
        {
            type: 'STAGE_TYPE',
            data: {
                field: 'VALUE'
            }
        }
    ],
    documents: [
        {
            type: 'DOCUMENT_TYPE',
            name: 'DOCUMENT_NAME',
            status: 'DOCUMENT_STATUS',
            uuid: 'UUID',
            created: 'DOCUMENT_TIMESTAMP'
        }
    ]
};

jest.mock('../../common/forms/form.jsx', () => 'MOCK_FORM');
jest.mock('axios', () => ({
    get: jest.fn((url) => {
        switch (url) {
        case '/case/summary/api':
            return Promise.resolve({
                data: { summary: { mockCaseData } }
            });
        default:
            return Promise.reject({
                response: {
                    data: {}
                }
            });
        }
    })
}));

import axios from 'axios';

describe('Summary page component', () => {

    const mockDispatch = jest.fn();
    let mockMatch = {};

    beforeEach(() => {
        mockDispatch.mockReset();
        mockMatch = {
            url: '/case/summary'
        };
    });

    it('should render with default props', () => {
        const outer = shallow(<Summary match={mockMatch} />);
        const Children = outer.props().children;
        const wrapper = mount(
            <MemoryRouter>
                <Children dispatch={mockDispatch} />
            </MemoryRouter>
        );
        expect(wrapper).toBeDefined();
        expect(wrapper.find('Summary')).toMatchSnapshot();
        expect(axios.get).toHaveBeenCalled();
    });

});