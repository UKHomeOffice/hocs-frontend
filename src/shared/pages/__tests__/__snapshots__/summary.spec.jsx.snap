// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Summary page component should render with default props 1`] = `null`;
